cfgmgr/vlanmgr.cpp

```diff
@@ -0,0 +1,488 @@
+    EXEC_WITH_ERROR_THROW(cmd.str(), res);
+
+    return true;
+}
+
+bool VlanMgr::removeHostVlanMember(int vlan_id, const string &port_alias)
+{
+    stringstream cmd;
+    string res;
+
+    cmd << BRIDGE_CMD << " vlan del vid " << vlan_id << " dev " << port_alias;
+    EXEC_WITH_ERROR_THROW(cmd.str(), res);
+
+    cmd.str("");
+    // When port is not member of any VLAN, it shall be detached from Dot1Q bridge!
+    cmd << BRIDGE_CMD << " vlan show dev " << port_alias << " | " << GREP_CMD << " None";
+    EXEC_WITH_ERROR_THROW(cmd.str(), res);
+    if (!res.empty())
+    {
+        cmd.str("");
+        cmd << IP_CMD << " link set " << port_alias << " nomaster";
+        EXEC_WITH_ERROR_THROW(cmd.str(), res);
+    }
+
+    return true;
+}
+
+bool VlanMgr::isVlanMacOk()
+{
+    return !(!gMacAddress);
+}
+
+void VlanMgr::doVlanTask(Consumer &consumer)
+{
+
+    if (!isVlanMacOk())
+    {
+        SWSS_LOG_DEBUG("VLAN mac not ready, delaying VLAN task");
+        return;
+    }
+    auto it = consumer.m_toSync.begin();
+
+    while (it != consumer.m_toSync.end())
+    {
+        auto &t = it->second;
+
+        string key = kfvKey(t);
+
+        /* Ensure the key starts with "Vlan" otherwise ignore */
+        if (strncmp(key.c_str(), VLAN_PREFIX, 4))
+        {
+            SWSS_LOG_ERROR("Invalid key format. No 'Vlan' prefix: %s", key.c_str());
+            it = consumer.m_toSync.erase(it);
+            continue;
+        }
+
+        int vlan_id;
+        vlan_id = stoi(key.substr(4));
+
+        string vlan_alias, port_alias;
+        string op = kfvOp(t);
+
+        if (op == SET_COMMAND)
+        {
+            string admin_status;
+            uint32_t mtu = 0;
+            vector<FieldValueTuple> fvVector;
+            string members;
+
+            /* Add host VLAN when it has not been created. */
+            if (m_vlans.find(key) == m_vlans.end())
+            {
+                addHostVlan(vlan_id);
+            }
+
+            /* set up host env .... */
+            for (auto i : kfvFieldsValues(t))
+            {
+                /* Set vlan admin status */
+                if (fvField(i) == "admin_status")
+                {
+                    admin_status = fvValue(i);
+                    setHostVlanAdminState(vlan_id, admin_status);
+                    fvVector.push_back(i);
+                }
+                /* Set vlan mtu */
+                else if (fvField(i) == "mtu")
+                {
+                    mtu = (uint32_t)stoul(fvValue(i));
+                    /*
+                     * TODO: support host VLAN mtu setting.
+                     * Host VLAN mtu should be set only after member configured
+                     * and VLAN state is not UNKOWN.
+                     */
+                    SWSS_LOG_DEBUG("%s mtu %u: Host VLAN mtu setting to be supported.", key.c_str(), mtu);
+                    fvVector.push_back(i);
+                }
+                else if (fvField(i) == "members@") {
```

Comment:
in the new configdb schema, we have VLAN_MEMBER table, we should remove the members@ attribute in VLAN table, otherwise, there cloud potentially be two sources of truth.

We do not need to do this change now. But, please do it in later PR.

Reply:
Ok, will remove it once minigraph.py update for the member processing is committed. 

https://github.com/Azure/sonic-buildimage/pull/1063/files#diff-18b32e4dc045fa3d1c5941a732124c89